src/nlr/Layer.cpp

```diff
@@ -879,38 +879,43 @@ void Layer::computeSymbolicBoundsForWeightedSum()
         */
 
-        for ( unsigned i = 0; i < _inputLayerSize; ++i )
-        {
-            for ( unsigned j = 0; j < _size; ++j )
-            {
-                if ( _eliminatedNeurons.exists( j ) )
-                    continue;
-
-                for ( unsigned k = 0; k < sourceLayerSize; ++k )
-                {
-                    _symbolicLb[i * _size + j] +=
-                        sourceLayer->getSymbolicUb()[i * sourceLayerSize + k] *
-                        _layerToNegativeWeights[sourceLayerIndex][k * _size + j];
-
-                    _symbolicLb[i * _size + j] +=
-                        sourceLayer->getSymbolicLb()[i * sourceLayerSize + k] *
-                        _layerToPositiveWeights[sourceLayerIndex][k * _size + j];
-
-                    _symbolicUb[i * _size + j] +=
-                        sourceLayer->getSymbolicUb()[i * sourceLayerSize + k] *
-                        _layerToPositiveWeights[sourceLayerIndex][k * _size + j];
-
-                    _symbolicUb[i * _size + j] +=
-                        sourceLayer->getSymbolicLb()[i * sourceLayerSize + k] *
-                        _layerToNegativeWeights[sourceLayerIndex][k * _size + j];
+        Map<unsigned, double> _eliminatedNeuronsLB;
+        Map<unsigned, double> _eliminatedNeuronsUB;
+        unsigned index;
+        for ( const auto &eliminated : _eliminatedNeurons ) {
+                for ( unsigned i = 0; i < _inputLayerSize; ++i ) {
+                    index = i * _size + eliminated.first;
+                    printf("_symbolicLB[%u] = %f\n", index, _symbolicLb[index]);
+                     _eliminatedNeuronsLB[index] = _symbolicLb[index];
+                     _eliminatedNeuronsUB[index] = _symbolicUb[index];
                 }
-            }
+        }
+
+        matrixMultiplication( sourceLayer->getSymbolicUb(), _layerToPositiveWeights[sourceLayerIndex],
+                              _symbolicUb, _inputLayerSize,
+                              sourceLayerSize, _size);
+        matrixMultiplication( sourceLayer->getSymbolicLb(), _layerToNegativeWeights[sourceLayerIndex],
+                              _symbolicUb, _inputLayerSize,
+                              sourceLayerSize, _size);
+        matrixMultiplication( sourceLayer->getSymbolicLb(), _layerToPositiveWeights[sourceLayerIndex],
+                              _symbolicLb, _inputLayerSize,
+                              sourceLayerSize, _size);
+        matrixMultiplication( sourceLayer->getSymbolicUb(), _layerToNegativeWeights[sourceLayerIndex],
+                              _symbolicLb, _inputLayerSize, sourceLayerSize,
+                              _size);
+
+        for ( const auto &eliminatedLB : _eliminatedNeuronsLB) {
+            _symbolicLb[eliminatedLB.first] = eliminatedLB.second;
+        }
+        for ( const auto &eliminatedUB : _eliminatedNeuronsUB) {
+            _symbolicUb[eliminatedUB.first] = eliminatedUB.second;
         }
```

Comment:
I think storing and restoring _symbolicLb and _symbolicUb for eliminated variables is not needed. In the base code, these values are zeroed out at the beginning of the function, and are never changed. So, because now blas might change these values, we indeed need to restore them, but this amounts to just zeroing them out again.